### PR TITLE
Make all fns in ethereum-provider arrow fns

### PR DESCRIPTION
**Pull request by hkirat**
Closes https://github.com/coral-xyz/backpack/issues/1044
The PR converts `handleEthAccounts` to an arrow fn as it was being bound to this before https://github.com/coral-xyz/backpack/pull/925 . 

Wasn't able to reproduce the error locally but I think this should fix it. @tomlinton could you help me confirm the fix if you can still recreate this locally?

### CODE REVIEW

File: packages/provider-core/src/provider-ethereum.ts [hunk 1]
```diff
@@ -425,12 +425,12 @@ export class ProviderEthereumInjection extends PrivateEventEmitter {
   /**
    * Handle eth_accounts requests
    */
-  async #handleEthAccounts() {
```
Comment (armaniferrante) [inline packages/provider-core/src/provider-ethereum.ts:428]: If we need to do this do we need to make everything else arrow functions? E.g. `handleAccountsChanged`?

Comment (hkirat) [reply packages/provider-core/src/provider-ethereum.ts:428]: No harm in doing that. Right now I kept it backward compatible with the list [here](https://github.com/coral-xyz/backpack/pull/925/files#diff-62fce178271dfd7972ea13095782602544cec13ad12014330d08245bf4d75bdaL146). 

I'll make all of em arrow.

**Comment by hkirat** [comment]
Any objections to converting all of the fns in all providers to arrow fns?
No real harm in doing it, no immediate benefit either unless fns inside them are called out of context

**Comment by tomlinton** [comment]
No objections, this seems to fix it from my local testing. I feel like prior to this we had autoreconnect working here. I might be wrong, can deal with it as a separate issue anyway.

Here is the repro if you want to try it:

https://user-images.githubusercontent.com/489202/195196421-fa3495ce-70b5-4107-80aa-c46434186115.mov

**Review by tomlinton — APPROVED** [review]

**Pull request merged**